drivers/display/display_st7789v.c

```diff
@@ -254,11 +255,48 @@ static int st7789v_set_pixel_format(const struct device *dev,
 static int st7789v_set_orientation(const struct device *dev,
 			    const enum display_orientation orientation)
 {
-	if (orientation == DISPLAY_ORIENTATION_NORMAL) {
-		return 0;
+	struct st7789v_data *data = (struct st7789v_data *)dev->data;
+	uint8_t tx_data = 0;
+	uint16_t x_offset = 0;
+	uint16_t y_offset = 0;
+
+	switch (orientation)
+	{
+	case DISPLAY_ORIENTATION_NORMAL:
+		tx_data |= ST7789V_MADCTL_MV_NORMAL_MODE;
+		x_offset = data->x_offset;
+		y_offset = data->y_offset;
+		break;
+
+	case DISPLAY_ORIENTATION_ROTATED_90:
+		tx_data |= (ST7789V_MADCTL_MY_BOTTOM_TO_TOP | ST7789V_MADCTL_MV_REVERSE_MODE);
+		x_offset = data->y_offset;
+		y_offset = data->x_offset;
+		break;
+
+	case DISPLAY_ORIENTATION_ROTATED_180:
+		tx_data |= (ST7789V_MADCTL_MY_BOTTOM_TO_TOP | ST7789V_MADCTL_MX_RIGHT_TO_LEFT);
+		x_offset = data->x_offset;
+		y_offset = data->y_offset;
+		break;
```

Comment:
```suggestion
	case DISPLAY_ORIENTATION_ROTATED_90:
		tx_data |= (ST7789V_MADCTL_MY_BOTTOM_TO_TOP | ST7789V_MADCTL_MV_REVERSE_MODE);
		x_offset = 320 - data->height + data->y_offset;
		y_offset = 240 - data->width + data->x_offset;
		break;

	case DISPLAY_ORIENTATION_ROTATED_180:
		tx_data |= (ST7789V_MADCTL_MY_BOTTOM_TO_TOP | ST7789V_MADCTL_MX_RIGHT_TO_LEFT);
		x_offset = 240 - data->width + data->x_offset;
		y_offset = 320 - data->height + data->y_offset;
		break;
```
My display is 240x240. I need these extra constants to align the picture with the display.